spec/src/main/asciidoc/ch04-query-language.adoc

```diff
@@ -1494,8 +1495,14 @@ If the value of a
 `state_valued_path_expression` or `in_item` in an IN or NOT IN expression
 is `NULL` or unknown, the value of the expression is unknown.
```

Comment:
This should be probably updated to:
_If the value of a state_valued_path_expression, in_item, subquery or collection_valued_input_parameter in an IN or NOT IN expression..._
to cover all cases. WDYT?

Reply:
Can a subquery evaluate to `null`?

Also, I think I would say that a `collection_valued_input_parameter` can by _empty_, but never _null_.

Reply:
> Can a subquery evaluate to `null`?

IN expression definitely can (ie if set of values contains NULL) - at least at the SQL level; not sure about subquery

> 
> Also, I think I would say that a `collection_valued_input_parameter` can by _empty_, but never _null_.

Does sentence above - _There must be at least one element in the
comma separated list that defines the set of values for the `IN`
expression._ - allow empty?

Reply:
Honestly the more I look at that sentence of the spec, the less I understand it.

Consider:

```
gavin=# select 1 where 2 in (2, 3, null);
 ?column? 
----------
        1
(1 row)
```

```
mysql> select 1 where 2 in (2, 3, null);
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.00 sec)
```

It's not the case that a `null` item in the `in` list causes the whole `in` expression to evaluate to null (at least not on or and mysql). So I think that's simply a mistake in the legacy spec.

Reply:
> It's not the case that a null item in the in list causes the whole in expression to evaluate to null (at least not on postgres or mysql). So I think that's simply a mistake in the legacy spec.

try [this example](https://stackoverflow.com/questions/4539340/sql-select-with-in-subquery-returns-no-records-if-the-sub-query-contains-null/4539354) to notice probably not too intuitive behaviour depending on availability of NULL in the list of values.

wrt empty list:
```
mysql> select 1 where 1 in ();
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ')' at line 1
```

Reply:
@lukasj I just pushed a commit, WDYT?